pkg/graveler/graveler.go

```diff
@@ -1898,6 +1929,7 @@ func (g *Graveler) ListStaging(ctx context.Context, branch *Branch, batchSize in
 
 // listStagingArea Returns an iterator which is an aggregation of all changes on all the branch's staging area (staging + sealed)
 // for each key in the staging area it will return the latest update for that key (the value that appears in the newest token)
+// listStagingArea will not return changes that were already compacted and saved in the CompactedBaseMetaRangeID
 func (g *Graveler) listStagingArea(ctx context.Context, b *Branch, batchSize int) (ValueIterator, error) {
```

Comment:
To avoid mistakes...
```suggestion
func (g *Graveler) listStagingAreaWithCompaction(ctx context.Context, b *Branch, batchSize int) (ValueIterator, error) {
```

Reply:
did you mean `listStagingAreaWithoutCompaction`?